### ISSUE TITLE
Shared Project

### ISSUE THREAD

**Issue by VanHinthem** [yagasoft/DynamicsCrm-CodeGenerator#4]
Those this also work in a shared project type?

**Comment by yagasoft** [comment]
Yes, you can use source control to share the configuration used to generate the entities. It's stored as JSON, so it's easy to merge with other branches.

**Comment by VanHinthem** [comment]
That not what I meant, I meant a "shared project type" in visual studio.

**Comment by yagasoft** [comment]
No, TT files are not supported in Shared Projects.

What you can do is add the TT file to another project as a container, add a symbolic link of the generated files in the 'shared project' folder in Windows pointing to the original files in the other project, and then add it to the 'shared project' in Visual Studio.

Windows Explorer:
- ModelProject
- - CrmSchema.tt
- - CrmSchema.cs
- - CrmSchemaBase.cs
- SharedProject
- - [Symbolic Link] CrmSchema.cs
- - [Symbolic Link] CrmSchemaBase.cs
- PluginProject
- - [Reference] SharedProject
Add the linked files to the SharedProject.

You could add the generated files as a direct link in the plugin project, but then you have to add all the generated files manually to every project, which might not be the most convenient.

I highly recommend that you generate a schema file per plugin project; for web services, you can use a single schema file.

**Comment by AngelRodriguez8008** [comment]
Hi @yagasoft,
Hi @VanHinthem,

I implemented a solution for this Issue, i can upload it in my Fork.

![image](https://user-images.githubusercontent.com/27274469/99155482-8a3ca400-26b8-11eb-917d-026bab4d0cfa.png)

br

**Comment by yagasoft** [comment]
Hello, @AngelRodriguez8008 ,

It would be great if you would create a pull request, please. This project structure is used frequently and would benefit many people indeed.

**Comment by AngelRodriguez8008** [comment]
Hi @yagasoft,

I did it one year ago or more, i remember your version was too complex to make the Changes that I needed & i decided to use original code from xairrick.

I have no time at the moment to make a Pull Request in your Fork, but I guest you can take a look what I did in the follow code and redo it.

[CrmCodeGenerator.VSPackage/CrmCodeGenerator.VSPackage.vsct](https://github.com/xairrick/CrmCodeGenerator/compare/master...AngelRodriguez8008:develop/Angel.Rodriguez#diff-40e99c004ea1f3b4ec1a4f4e88b3ff87fdc3daed0d826e853fd1983f0c3a9798R72) at Line 72

[CrmCodeGenerator.VSPackage/Commands/ApplyCustomTool.cs](https://github.com/xairrick/CrmCodeGenerator/compare/master...AngelRodriguez8008:develop/Angel.Rodriguez#diff-d74ba8eacdeb56fa1868e42beabee7f5739c632ebd3fe27ad5ed23f1ef77bbec)

I can put it in my todo-list but it can take some time (months), until it get done.

br

**Comment by yagasoft** [comment]
Thank you, @AngelRodriguez8008 .

**Comment by AngelRodriguez8008** [comment]
Your welcome @yagasoft ! Feel free to contact me if you need some help.
